src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/parameterassignment/InputParameterAssignmentWithUnchecked.java

```diff
@@ -40,7 +40,32 @@ void foo5(int EXPR) {
         int i = EXPR;
     }
 
+    SomeInterface obj = q -> q++; // violation
+    SomeInterface obj2 = (int q) -> q += 12; // violation
+    SomeInterface obj3 = (w) -> w--; // violation
+    AnotherInterface obj4 = (int q, int w) -> obj.equals(obj2);
+    AnotherInterface obj5 = (q, w) -> w = 14; // violation
+
+    void method() {
+        int q = 12;
+        SomeInterface obj = (d) -> {
+            SomeInterface b = (c) -> obj2.equals(obj4);
+            int c = 12;
+            c++;
+            SomeInterface r = (field) -> this.field++;
+            d -= 10; // violation
```

Comment:
please also add cases with annotated parameter.
You can see more examples for lambdas in https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/master/sevntu-checks/src/test/resources/com/github/sevntu/checkstyle/checks/coding/InputPreferMethodReferenceCheck1.java and same inputs for this check, some of them can be useful

Reply:
Added some more inputs.